### ISSUE TITLE
SSBU: Unable to write to this amiibo.

### ISSUE THREAD

**Issue by Mou-Ikkai** [Ryujinx/Ryujinx#2457]
### What's the issue you encountered?
After scanning my Joker amiibo and training him, when I went to rescan the Joker Amiibo, I got an error: "The data on this amiibo may have been overwritten by another console.  Unable to write to this amiibo."

### How can the issue be reproduced?
Train an amiibo, save it, close the emulator, reopen, train the same amiibo to a higher level, saving should get the error. (Make sure it is the same amiibo! Can't overwrite/resave it if you use a different one!)

### Environment?

- Ryujinx version: 1.0.6956</br>

- Game version: 12.0.0</br>
- System Specs:
  - OS: *Windows 10 Pro v19043.1081*
  - CPU: *Ryzen 9 3900X*
  - GPU: *RX 5700XT 8GB*
  - RAM: *32GB*
- Applied Mods :  None.

### Additional context?
![image](https://user-images.githubusercontent.com/45863583/125020958-c0fc6b80-e047-11eb-8947-8213eebe18d4.png)
![image](https://user-images.githubusercontent.com/45863583/125021012-d5d8ff00-e047-11eb-8d22-0c366832f331.png)
![image](https://user-images.githubusercontent.com/45863583/125021033-e0939400-e047-11eb-92d4-2ff2da489682.png)

**Comment by Shahil-Ayato** [comment]
This should be reported under [This ](https://github.com/Ryujinx/Ryujinx/issues/2122) issue.

**Comment by AcK77** [comment]
I'm not able to reproduce the issue, have you checked "Random Tag UUID" the first time? It could cause this type of issue and that's why it's disabled as default.

**Comment by Mou-Ikkai** [comment]
> I'm not able to reproduce the issue, have you checked "Random Tag UUID" the first time? It could cause this type of issue and that's why it's disabled as default.

No, I never used the Random Tag UUID. I've just been training my Amiibo over time, and this occured.

**Comment by Mou-Ikkai** [comment]
Occured again with another amiibo.
![image](https://user-images.githubusercontent.com/45863583/132067318-8305c9e9-97c9-47da-93f5-ccb58db61909.png)
![image](https://user-images.githubusercontent.com/45863583/132067330-480d76ae-9876-490d-818a-61ccb1d6babc.png)
![image](https://user-images.githubusercontent.com/45863583/132067341-2f799ef8-cc7c-4af3-9115-be2008e76605.png)

**Comment by Mou-Ikkai** [comment]
It seems random, though. Because I can scan my Joker amiibo constantly no issue. But now Little Mac, a new amiibo, had this issue on the 4th scan it seems? And you can't scan again to fix it either.

**Comment by Mou-Ikkai** [comment]
![image](https://user-images.githubusercontent.com/45863583/132249850-251f2ee2-33e8-467c-8d6b-558b8cb82dd7.png)
![image](https://user-images.githubusercontent.com/45863583/132249877-21acb8d0-0788-437d-b01d-71f7faca0c3e.png)
![image](https://user-images.githubusercontent.com/45863583/132249885-40faa34a-439b-432d-9e11-57b59a5ac12e.png)
At this point, I believe it might occur if you write two or three times to the amiibo without restarting the emulator, since this was another new amiibo.

**Comment by Mou-Ikkai** [comment]
![image](https://user-images.githubusercontent.com/45863583/140428300-97d3e62d-cdc5-4758-b43a-e3d43ad7d2bf.png)
![image](https://user-images.githubusercontent.com/45863583/140428447-2fa70bde-e14b-4f2e-b255-8f1ee25fccd2.png)
![image](https://user-images.githubusercontent.com/45863583/140428472-a82595b6-06ce-4486-9bc0-3f270ee7aae9.png)
Always seems to be the 2nd or 3rd time that I try to write that this issue happens.

OS - Linux Mint 20.2

**Comment by Mou-Ikkai** [comment]
![image](https://user-images.githubusercontent.com/45863583/141666960-5a4e8af5-7b7f-49af-8c7a-def528c0bcf5.png)
Happened again when saving Little Mac for the 2nd time.

**Comment by Mou-Ikkai** [comment]
It can be random it seems, but it seems to be that after a major data change (training your amiibos a few levels, then saving it, then doing a few more and saving again), can cause this issue.

**Comment by Mou-Ikkai** [comment]
Reproducable steps: Train a fresh amiibo to about level 20-30, save it. Then close Ryujinx, reopen it, then scan the amiibo again, train more levels, then save again... Or try to as this issue will occur.
![image](https://user-images.githubusercontent.com/45863583/141888468-b2feb5de-9d47-4afb-9634-7a6c4bc310d3.png)
![image](https://user-images.githubusercontent.com/45863583/141888573-9e6dd1c3-eb26-4df4-8116-0ae1da5f31e2.png)
![image](https://user-images.githubusercontent.com/45863583/141888589-fbb876a3-ff99-430d-a101-3bb080545b53.png)

**Comment by Mou-Ikkai** [comment]
So, I think it has something to do with the first save on the JSON file... If I go into Smash, make a new Amiibo, then play a few games and save, it's fine. That always happens. Now then, this is where it splits: If I delete the file, then repeat the process earlier, it's still fine! However, if I don't delete the file, then try to save again... That's when it gives the error.

**Comment by Mou-Ikkai** [comment]
*Also, this report is on my laptop:
Ryzen 5 5600H
3050Ti
16GB-3200DDR4*

**Comment by Rodrigosouzaspec** [comment]
I am with the same issue, did you find a way to solve it?

**Comment by Mou-Ikkai** [comment]
> I am with the same issue, did you find a way to solve it?

Nope...

**Comment by therealfreaksoul** [comment]
I might be very late to the party, but here is a step-by-step guide to save your Amiibos without having the error to ruin progress (it works for me at least)
1) Restart the game
2) Go to "Games & More"
3) Go to "Amiibo"
4) Scan your amiibo
5) Click on "Smash"
6) Play 2 games with your desired Ruleset
7) After those two games, you will be send back to the Stage selection menu
8) Go back until you are at the "Would you like to return to the amiibo menu?" Screen
9) Click yes
10) Click on "Put Away"
11) Scan your amiibo (This should safe your amiibo)
12) Then, whilst on the "Touch an amiibo to the NFC Touchpoint" Screen, Scan your amiibo again
13) If you have done everything right so far, the game gifts you something either spirit points or a snack
14) Press next
15) Click "Put away"
16) Scan your amiibo
17) you will get the error message "The data on this amiibo may have been [...]"
18) Press "B" so you will get the "Are you sure you want to quit without saving [...]" prompt
19) click yes
20) repeat from step 4)

A few side notes;
- You might be able to play more than 2 games, i just haven't tried to do more than 2
- It seems like the collection of the gift might have something to do with the error but who knows
- While it is convoluted, it does at least work

**Comment by therealfreaksoul** [comment]
Ok i tested a little bit more and the step by step guide is pretty unnecessarily long. **If you get gifts after scanning your amiibo click put away to get the error message**. Quit out of the amiibo menu and scan it again. Go to smash and do whatever
